code/modules/client/preferences.dm

```diff
@@ -1230,16 +1254,22 @@ var/global/list/special_role_times = list( //minimum age (in days) for accounts
 							b_markings = hex2num(copytext(new_markings, 6, 8))
 
 				if("m_style")
-					if((species in list("Unathi", "Vulpkanin", "Tajaran"))) // Species with markings
+					if((species in list("Unathi", "Vulpkanin", "Tajaran", "Machine"))) // Species with markings
 						var/list/valid_markings = list()
 						for(var/markingstyle in marking_styles_list)
 							var/datum/sprite_accessory/M = marking_styles_list[markingstyle]
 							if( !(species in M.species_allowed))
 								continue
 
+							if(species == "Machine") //Species that can use prosthetic heads.
+								if(!("head" in rlimb_data) && M.name != "None") //If the character can have prosthetic heads and they have the default head (with screen), no optic markings.
+									continue
+								else if(("head" in rlimb_data) && M.name == "None") //Otherwise, if they DON'T have the default head and since they must have optics, give them a list of styles excluding the "None" option.
+									continue
+
 							valid_markings[markingstyle] = marking_styles_list[markingstyle]
 
-						var/new_marking_style = input(user, "Choose the style of your character's body markings:", "Character Preference") as null|anything in valid_markings
+						var/new_marking_style = input(user, "Choose the style of your character's markings:", "Character Preference") as null|anything in valid_markings

```

Comment:
Changing this to
```
var/new_marking_style = input(user, "Choose the style of your character's markings:", "Character Preference", m_style) as null|anything in valid_markings
```
should make the current markings option be the default/initially selected one, which is a good quality of life thing.